File: KantaroBLL/Canzone/Canzoni.vb
Public Class Canzoni

    Public Class CanzoniItem
        Public Property FilePath As String
        Public Property Title As String
    End Class

    Private _canzoni As New List(Of CanzoniItem)

    Public Sub New()

    End Sub

    Public Sub New(fileKantoj As String)
        Dim xDoc = XDocument.Load(fileKantoj)
        _canzoni.AddRange(FromXDocument(xDoc, IO.Path.GetDirectoryName(fileKantoj)).Items)
    End Sub

    ''' <summary>
    ''' Restituisce la classe Canzoni relativa all'xml del file kantoj
    ''' </summary>
    ''' <param name="document">Oggetto XDocument della lista di canzoni</param>
    ''' <param name="basePath">Percorso del file system dove è contenuto il file d'origine</param>
    ''' <returns></returns>
    Public Shared Function FromXDocument(document As XDocument, basePath As String) As Canzoni
        Dim kj As New Canzoni()
        Dim canti = From k In document...<kanto>
                    Let filePath = IO.Path.Combine(basePath, k.Value)
                    Select New CanzoniItem() With {.FilePath = filePath, .Title = k.@title}
        kj._canzoni.AddRange(canti)
        Return kj
    End Function

    Public Function ToXDocument() As XDocument
        Dim doc = <?xml version="1.0" encoding="utf-8"?><kantoj/>
        For Each c In Items
            Dim k = <kanto><%= c.FilePath %></kanto>
            If Not String.IsNullOrEmpty(c.Title) Then
                k.@title = c.Title
            End If
            If doc.Root.LastNode Is Nothing Then
                doc.Root.AddFirst(k)
            Else
                doc.Root.LastNode.AddAfterSelf(k)
            End If
        Next
        Return doc
    End Function

    Public ReadOnly Property Items As IEnumerable(Of CanzoniItem)
        Get
            Return _canzoni
        End Get
    End Property

    Public Sub Add(pathCanzone As String, title As String)
        _canzoni.Add(New CanzoniItem() With {.FilePath = pathCanzone, .Title = title})
    End Sub

    Public Sub Add(item As CanzoniItem)
        _canzoni.Add(item)
    End Sub
End Class

File: Enzo.Music/Canzone/Canzoni.cs
using System.Xml.Linq;

namespace Enzo.Music;
public class Canzoni
{
    public class CanzoniItem
    {
        public string FilePath;
        public string Title;

        internal CanzoniItem(string filePath, string title)
        {
            FilePath = filePath;
            Title = title;
        }
    }
    private readonly List<CanzoniItem> canzoni;
    public IEnumerable<CanzoniItem> Items => canzoni;

    public Canzoni()
    {
        canzoni = new List<CanzoniItem>();
    }

    public Canzoni(string fileKantoj) : this()
    {
        var xDoc = XDocument.Load(fileKantoj);
        canzoni.AddRange(FromXDocument(xDoc, Path.GetDirectoryName(fileKantoj)).Items);
    }

    public void Add(string pathCanzone, string title)
    {
        canzoni.Add(new CanzoniItem(pathCanzone, title));
    }

    public void Add(CanzoniItem item)
    {
        canzoni.Add(item);
    }

    /// <summary>
    /// Restituisce la classe Canzoni relativa all'xml del file kantoj
    /// </summary>
    /// <param name="document">Oggetto XDocument della lista di canzoni</param>
    /// <param name="basePath">Percorso del file system dove è contenuto il file d'origine</param>
    /// <returns></returns>
    public static Canzoni FromXDocument(XDocument document, string? basePath)
    {
        Canzoni kj = new();
        var canti = from k in document.Descendants("kanto")
                    let filePath = Path.Combine(basePath ?? string.Empty, k.Value)
                    select new CanzoniItem(filePath, k.Attribute("title")?.Value ?? string.Empty);
        kj.canzoni.AddRange(canti);
        return kj;
    }

    public XDocument ToXDocument()
    {
        XDocument doc = new(new XDeclaration("1.0", "utf-8", "yes"), new XElement("kantoj"));
        foreach (var c in Items)
        {
            XElement k = new("kanto") { Value = c.FilePath };
            if (!string.IsNullOrEmpty(c.Title))
                k.SetAttributeValue("title", c.Title);
            if (doc.Root?.LastNode is null)
                doc.Root?.AddFirst(k);
            else
                doc.Root.LastNode.AddAfterSelf(k);
        }
        return doc;
    }
}